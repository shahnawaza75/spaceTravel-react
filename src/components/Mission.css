.mission-title {
  display: grid;
  grid-template-columns: 10.9% 65% 12.09% 8%;
  grid-template-rows: auto;
  align-items: center;
  border: 1px solid #d8d8d8;
  height: 48px;
  padding-left: 4px;
}

.main {
  margin: 20px 50px 20px;
}

.title-desc {
  border-left: 1px solid #d8d8d8;
  border-right: 1px solid #d8d8d8;
  height: 48px;
  display: flex;
  align-items: center;
  padding-left: 4px;
}

.title-status {
  border-right: 1px solid #d8d8d8;
  height: 48px;
  display: flex;
  align-items: center;
  margin-left: 4px;
}
