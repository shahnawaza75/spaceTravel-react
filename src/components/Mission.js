import React from 'react';
import { useSelector } from 'react-redux';
import MissionList from './MissionList';
import './Mission.css';

const Mission = () => {
  const missions = useSelector((state) => state.mission);

  return (
    <div>
      <div className="main">
        <div className="mission-title">
          <h3 className="title-mission">Mission</h3>
          <h3 className="title-desc">Description</h3>
          <h3 className="title-status">Status</h3>
        </div>
        {missions.map((mission) => (
          <MissionList key={mission.id} list={mission} />
        ))}
      </div>
    </div>
  );
};

export default Mission;
