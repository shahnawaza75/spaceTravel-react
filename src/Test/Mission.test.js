import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/index';
import Mission from '../components/Mission';

describe('Test the navigator', () => {
  it('Navigation is rendered', () => {
    const missions = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Mission />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(missions).toMatchSnapshot();
  });
});
