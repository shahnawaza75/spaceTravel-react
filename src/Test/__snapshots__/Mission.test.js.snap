// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the navigator Navigation is rendered 1`] = `
<div>
  <div
    className="main"
  >
    <div
      className="mission-title"
    >
      <h3
        className="title-mission"
      >
        Mission
      </h3>
      <h3
        className="title-desc"
      >
        Description
      </h3>
      <h3
        className="title-status"
      >
        Status
      </h3>
    </div>
  </div>
</div>
`;
